ixdiagnose/utils/command.py

```diff
@@ -0,0 +1,33 @@
+import subprocess
+
+from typing import Optional, Union
+
+from ixdiagnose.exceptions import CallError
+
+from .run import run
+
+
+class Command:
+
+    def __init__(
+        self, command: Union[str, list], description: str, shell: bool = False,
+        serializeable: bool = True, safe_returncodes: list = None, env: Optional[dict] = None,
+        max_lines: Optional[int] = None,
+    ):
+        self.command: Union[str, list] = command
+        self.description: str = description
+        self.env: Optional[dict] = env
+        self.max_length: Optional[int] = max_lines
+        self.shell: bool = shell
+        self.serializeable: bool = serializeable
+        self.safe_returncodes: list = safe_returncodes or [0]
+        if self.shell and not isinstance(self.command, str):
```

Comment:
Can we just determine `shell` based on the `command` type? Looks excessive to force user to specify `shell` explicitly`.